src/test/java/DirectionsTest.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import com.google.maps.DirectionsApplication;
+import com.google.maps.DirectionsConsumer;
+import com.google.maps.DirectionsService;
+import com.google.maps.DirectionsServiceInjector;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class DirectionsTest {
+
+  private DirectionsServiceInjector injector;
+  private final String ORIGIN = "University of Waterloo";
+  private final String DESTINATION = "Wilfrid Laurier University";
+  private final List<String> ONE_WAYPOINT = Arrays.asList("University of Waterloo Place");
+  private final String API_KEY = "SAMPLE_API_KEY";
+
+  @Before
+  public void init() {
+    // Mock the injector with an anonymous class
+    injector =
+        new DirectionsServiceInjector() {
+          @Override
+          public DirectionsConsumer getConsumer() {
+            // Mock the Directions service
+            return new DirectionsApplication(
+                new DirectionsService() {
+                  @Override
+                  public String getDirections(String uri) {
+                    System.out.println("Mock Directions Service Implementation");
+                    return null;
+                  }
+                });
+          }
+        };
+  }
+
+  @After
+  public void tear() {
+    injector = null;
+  }
+
+  @Test
```

Comment:
I'm not sure if this test class is actually needed. 

All this test does is creates a mock (which always returns null), and then tests that the mock will always return null. 

The classes that only do trivial GET and POST operations don't need to be tested - we should be able to assume the response is as desired from Google. Any manipulation of these responses should be tested, and done in a seperate public function so they can be tested/mocked as appropriate. 

Don't test responses, but do test transformations

Reply:
I absolutely agree, I have changed the new tests such that it tests the `.json` transformations instead of the HTTP requests. Thanks!

Reply:
`RESOLVED`